schema/schema_test.go

```diff
@@ -0,0 +1,389 @@
+	// invalid
+	type empty struct {
+	}
+
+	// invalid
+	type noPK struct {
+		Key1 string `json:"key_1"`
+	}
+
+	// valid. implicit one part primary key with index 1
+	type pk struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key"`
+	}
+
+	// invalid. index starts from 1
+	type pk0 struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key:0"`
+	}
+
+	// valid. optional index specified
+	type pk1 struct {
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	// valid. composite primary key
+	type pk2 struct {
+		Key2 string `json:"key_2" tigrisdb:"primary_key:2"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	type pkDup struct {
+		Key2 string `json:"key_2" tigrisdb:"primary_key:1"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	// invalid. promary key index greater then number of fields
+	type pkOutOfBound struct {
+		Key3 string `json:"key_3" tigrisdb:"primary_key:3"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+	}
+
+	// invalid. gap in primary key indexes
+	type pkGap struct {
+		Key3  string `json:"key_3" tigrisdb:"primary_key:3"`
+		Key1  string `json:"key_1" tigrisdb:"primary_key:1"`
+		Data1 string
+	}
+
+	type pkInvalidType struct {
+		Key2 bool `json:"key_2" tigrisdb:"primary_key:1"`
+	}
+
+	type pkInvalidTag struct {
+		Key2 bool `json:"key_2" tigrisdb:"primary_key:1:a"`
+	}
+
+	type pk3 struct {
+		Key2 string `json:"key_2" tigrisdb:"primary_key:2"`
+		Key1 string `json:"key_1" tigrisdb:"primary_key:1"`
+		Key3 string `json:"key_3" tigrisdb:"primary_key:3"`
+	}
+
+	type subSubStructPK struct {
+		Data1  int    `tigrisdb:"-"`
+		Field1 string `json:"ss_field_1" tigrisdb:"primary_key:1"`
+	}
+
+	type subStructPK struct {
+		Field1 string `json:"field_1"`
+		Nested subSubStructPK
+	}
+
+	type subSubStruct struct {
+		Data1  int    `tigrisdb:"-"`
+		Field1 string `json:"ss_field_1"`
+	}
+
+	type subStruct struct {
+		Field1 string `json:"field_1"`
+		Nested subSubStruct
+	}
+
+	type allTypes struct {
+		tm     time.Time
+		tmPtr  *time.Time
+		Int8   int8   `json:"int_8"`
+		Int16  int16  `json:"int_16"`
+		Int32  int32  `json:"int_32"`
+		Uint8  uint8  `json:"uint_8"`
+		Uint16 uint16 `json:"uint_16"`
+		Uint32 uint32 `json:"uint_32"`
+
+		Int64 int64 `json:"int_64"`
+		//Uint64 uint64 `json:"uint_64"`
+		Int  int  `json:"int_1"`
+		Uint uint `json:"uint_1"`
+
+		Bytes  []byte  `json:"bytes_1"`
+		BytesA [4]byte `json:"bytes_2"`
+
+		Float32 float32 `json:"float_32"`
+		Float64 float64 `json:"float_64"`
+
+		Bool bool `json:"bool_1"`
+
+		String string `json:"string_1" tigrisdb:"primary_key:2"`
+
+		Data1  subStructPK          `json:"data_1"`
+		Slice1 []string             `json:"slice_1"`
+		Arr1   [3]string            `json:"arr_1"`
+		Map    map[string]string    `json:"map_1"`
+		Slice2 []subStruct          `json:"slice_2"`
+		Map2   map[string]subStruct `json:"map_2"`
+		Data2  subStructPK          `json:"data_2" tigrisdb:"primary_key:3,-"` // should be skipped
```

Comment:
what will be the use case of this? `tigrisdb:"primary_key:3,-"` // should be skipped`

Reply:
This just tests that skip is recognised in multipart tag and have precedence.